src/Nethermind/Nethermind.AccountAbstraction/Source/UserOperationPool.cs

```diff
@@ -247,7 +274,12 @@ public ResultWrapper<Keccak> AddUserOperation(UserOperation userOperation)
                 if (_userOperationSortedPool.TryInsert(userOperation.RequestId!, userOperation))
                 {
                     Metrics.UserOperationsPending++;
-                    _paymasterThrottler.IncrementOpsSeen(userOperation.Paymaster);
+                    UInt256 baseFee = CalculateCurrentBaseFee();
```

Comment:
Calculating twice - here and in Validate, calculate once before validate, and pass it to validate.